### PR TITLE
Add elementwise min grad NPU OP

### PR DESCRIPTION
**Pull request by andyjiang1116**
### PR types

New features
### PR changes

OPs
### Describe

[NPU] Support npu op elementwise_min and elementwise_min_grad (with broadcast)
CANN OP elementwise_min_grad broadcast only support axis=-1
unittest result:
image
GLOG:
image

### CODE REVIEW

**Review by zhhsplendid — COMMENTED** [review]
LGTM for ShareDataWith

**Review by zhupengyang — DISMISSED** [review]
LGTM

**Comment by andyjiang1116** [comment]
> LGTM for ShareDataWith

**Comment by andyjiang1116** [comment]
> LGTM for ShareDataWith

thanks！

**Comment by andyjiang1116** [comment]
> LGTM

thanks！

**Review by zhangting2020 — DISMISSED** [review]
LGTM for skip_check_grad_ci

**Review by MissPenguin — DISMISSED** [review]

File: paddle/fluid/framework/ir/CMakeLists.txt [hunk 1]
```diff
@@ -59,7 +59,7 @@ cc_library(coalesce_grad_tensor_pass SRCS coalesce_grad_tensor_pass.cc DEPS grap
 
 pass_library(graph_to_program_pass base)
 pass_library(graph_viz_pass base)
-pass_library(lock_free_optimize_pass base)
+pass_library(lock_free_optimize_pass base DEPS string_helper)
```
Comment (qili93) [inline paddle/fluid/framework/ir/CMakeLists.txt]: 这个的修复PR已经合入了，需要删掉

**Comment by qili93** [comment]
https://github.com/PaddlePaddle/Paddle/pull/34057 已经合入，部分代码与这个PR重复，请更新一下。

File: paddle/fluid/framework/ir/CMakeLists.txt (hunk 1 above)
Comment (andyjiang1116) [reply paddle/fluid/framework/ir/CMakeLists.txt]: 好的，谢谢

**Review by zhiqiu — APPROVED** [review]
LGTM for ShareDataWith

**Pull request closed without merging**